### ISSUE TITLE
Fix the spelling of Nicolson

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#15080]
As pointed out correctly by @olegrog in #15078, the proper spelling of Phyllis Nicolson is not "Nicholson". There are several more occurrences in the library worth fixing:
```
include/deal.II/algorithms/theta_timestepping.h:   * schemes, the Crank-Nicholson scheme and linear combinations of those. The
include/deal.II/algorithms/theta_timestepping.h:   * <li> #theta=½: Crank-Nicholson scheme
include/deal.II/algorithms/theta_timestepping.h:   * For fixed #theta, the Crank-Nicholson scheme is the only second order
examples/step-31/doc/intro.dox:evaluation of the residual is nothing else than a Crank-Nicholson scheme,
```

**Comment by olegrog** [comment]
It seems that I have already included all such fixes in #15078.

**Comment by marcfehling** [comment]
Closed by #15083.
